addons/mail/static/src/models/rtc/rtc.js

```diff
@@ -976,23 +1000,23 @@ function factory(dependencies) {
         }
 
         /**
-         * Sets the enabled property of the local audio track based on the
-         * current session state. And notifies peers of the new audio state.
+         * Sets the enabled property of the local audio track and notifies
+         * peers of the new state.
          *
          * @private
          */
         async _updateLocalAudioTrackEnabledState() {
             if (!this.audioTrack) {
                 return;
             }
-            this.audioTrack.enabled = !this.currentRtcSession.isMuted && this.currentRtcSession.isTalking;
+            this.audioTrack.enabled = !this.currentRtcSession.isMuteOrDeaf && this.currentRtcSession.isTalking;
             await this._notifyPeers(Object.keys(this._peerConnections), {
                 event: 'trackChange',
                 type: 'peerToPeer',
                 payload: {
                     type: 'audio',
                     state: {
-                        isTalking: this.currentRtcSession.isTalking && !this.currentRtcSession.isMuted,
+                        isTalking: this.audioTrack.enabled,
```

Comment:
Can't do the same change in other places? In particular `dataChannel.onopen`

Reply:
Or why this one doesn't write the `isTalking` field?
The inconsistency is strange.